app/src/main/java/mozilla/lockbox/support/FxASyncDataStoreSupport.kt

```diff
@@ -32,26 +32,26 @@ class FxASyncDataStoreSupport(
         }
 
         // val encryptionKey = randomUUID().toString()
-        val encryptionKey = generateRandomString(50)
+        val encryptionKey = generateEncryptionKey(256)
         preferences.putString(Constant.Key.encryptionKey, encryptionKey)
 
         encryptionKey
     }
 
-    /*
-     * This serves as a placeholder, and should definitely not be considered secure.
-     * TODO make this cryptographically interesting; add method to KeyStore/DataProtect
+    /**
+     * @param keyStrength The strength ofthe generated key in bits
      */
-    private fun generateRandomString(keyLength: Int): String {
-        val charPool = "0123456789ABCDEF".toCharArray()
-
-        val bytes = ByteArray(keyLength)
+    private fun generateEncryptionKey(keyStrength: Int): String {
+        val bytes = ByteArray(keyStrength / 8)
         val random = SecureRandom()
         random.nextBytes(bytes)
 
         return bytes
             .map {
-                charPool[(it.toInt() and 0xFF % charPool.size)]
+                val full = it.toInt()
+                val hi = (full and 0xf0) ushr 4
+                val lo = full and 0x0f
+                "%h%h".format(hi, lo)
```

Comment:
I just was checking to see if this formatted upper case or lower case (for https://github.com/mozilla/application-services/issues/499) and apparently %h isn't the hex formatter, it's the hashcode formatter. This is, of course, further argument that 499 is important, but it's worth noting that it won't be compatible in that way.

Reply:
🤦‍♂️  thanks for the catch.